SteamKit2/SteamKit2/Steam/Handlers/SteamAuthentication/SteamAuthentication.cs

```diff
@@ -281,13 +283,16 @@ public async Task<AuthPollResult> StartPolling( CancellationToken? cancellationT
 
                 while ( true )
                 {
-                    cancellationToken?.ThrowIfCancellationRequested();
-
-                    await Task.Delay( PollingInterval );
-
-                    cancellationToken?.ThrowIfCancellationRequested();
+                    if( cancellationToken is CancellationToken nonNullCancellationToken )
```

Comment:
This should fade away after making CancellationToken non-nullable as mentioned above.